### DISCUSSION TITLE
Embedded metadata for book collection

### DISCUSSION THREAD

**Discussion by kyledenlinger** [knowledgefutures/pubpub#2124] [Support]
It seems that the metadata PubPub exposes for some Book-type collections identifies* the collection itself as a journal article rather than a book, but the individual pubs are treated as book chapters. Example: [this collection](https://decarbonizingcharacter.pubpub.org/sp22) doesn't have much in it yet, but it should be a book. [This pub in that collection](https://decarbonizingcharacter.pubpub.org/pub/gratitudes-entanglement/release/3?readingCollection=f116f516) is identified as a chapter in the book, and the citation looks right.

*I'm going entirely on what my Zotero connector sees.

**Comment by tefkah** [comment]
I think it's because almost no `<meta>` tags are set for layout pages, I made a similar request here: https://github.com/pubpub/pubpub/discussions/1565#discussioncomment-3063822, would be really cool to see supported!

**Reply by gabestein** [reply]
Thanks for this! We'd only focused on Pub `<meta>` to date, but definitely see the value for Collection layouts, and shouldn't be too complex. We'll take a look.

**Reply by tefkah** [reply]
Has there been any movement on this? I would be happy to supply a PR for this if not!

**Reply by gabestein** [reply]
Hi Thomas — sorry it's taken longer that we'd like! If you feel up for it, please go for it! Much appreciated.

We do intend to expose this metadata, but something is happening I think with properly identifying collection types in https://github.com/pubpub/pubpub/blob/master/server/utils/ssr.tsx#L126.

So, I think it'll mostly be figuring out what's going on there and making sure the metadata displayed for collections is correct.

If you have any issues, let me know — happy to help out, but at a conference this week so may be a bit slow to respond.

P.S. -- I've been meaning to reach out, but we're working on an update to our DOI support that uses some of your work converting the crossref and datacite XSDs to typescript types. I'll send you a note via email :)

**Reply by tefkah** [reply]
Hi Gabe! No worries at all! 

I think I see the issue: the `generateMetadataComponents` does not get passed the `collection` here:
https://github.com/pubpub/pubpub/blob/bf844fe9bd00a50690197aed4bd1e8d3b27c7542/server/routes/collection.tsx#L119-L125

adding `collection,` will probably fix it, I'll make a quick PR!

And im happy to hear that the XSD typescript gen might be useful to you! It's in a bit of a rough state though as I'm reworking it (yet again) but I could definitely help you get set up more quickly if needed! The idea is to eventually create a proper Crossref XML generation library that is both typesafe, validated through zod, and provides an easy api to generate common types, but that might be a little ways off as I _really_ need to finish my thesis before running out of savings 😅. See https://github.com/tefkah/crossref-tools for the WIP

**Reply by tefkah** [reply]
See https://github.com/pubpub/pubpub/pull/2260

**Reply by gabestein** [reply]
Thanks! This is now live. As I'm not a big Zotero user, it would be great if you could take a look at a collection page and see if you get the metadata you're expecting. It looks to me like it's mostly working, as in the data is now populating, although Zotero recognizes it as just a "website" rather than a book or journal. But I've had trouble, reading docs, figuring out whether there is a way to save a site as a book based on HTML metadata.

**Reply by kyledenlinger** [reply]
@gabestein I'm by no means an expert, but I think the problem resides in this element: `<meta property="og:type" content="website"/>`

More on Open Graph object types here: https://ogp.me/#types

**Reply by tefkah** [reply]
Yeah if I change it to `content="website"` Zotero saves it as a book! There is some metadata missing though, most importantly Author, Year, and Edition

The Title also includes the community which is probably undesirable.

![image](https://user-images.githubusercontent.com/21983833/192360144-841cdba5-51c2-45ed-b22a-856f08e40e45.png)

**Reply by gabestein** [reply]
@kyledenlinger Thanks for the note! We can give it a shot for books, although I'm not 100% sure that's what Zenodo is looking at, and I don't what to do in the journal case. From what I can see, there is no type for journals (or conference proceedings for that matter). And actually, there isn't one for chapter, either -- we just use "article." Zotero recognizes something as a chapter I think entirely from the citation_inbook_title we add. Anyhow, we'll give the og type a try.

edit: @tefkah -- you're expecting a book type in that case, or something else?

**Reply by gabestein** [reply]
nevermind, I found a good mapper for Zotero. We just need to do a little more work in the book case to map these tags. Should be very straightforward. If you want to take it for a spin, @tefkah, go for it. Otherwise I'll try to get to it by eow.

https://github.com/zotero/translators/blob/master/Embedded%20Metadata.js

**Reply by tefkah** [reply]
Ah sorry I didn't see this, I gave it a shot manually here: https://github.com/pubpub/pubpub/pull/2265

**Reply by tefkah** [reply]
I think the problem is that it's using the information from the page, not the collection, otherwise most things are fine I think!

**Reply by gabestein** [reply]
No problem, and thank you! If and only if you feel up for it, you could expand that PR to accomadate this enhancement card: https://github.com/pubpub/pubpub/issues/2266

Otherwise, I can do it later this week.
